### ISSUE TITLE
[Search field] Errors occurring when search string is too long

### ISSUE THREAD

**Issue by FranECross** [department-of-veterans-affairs/va.gov-cms#15905]
## Description
Adrian Rollett reported errors from when VA.gov users submit a search string that's are too long.  Slack 🧵https://dsva.slack.com/archives/C05THHJHH2R/p1698674544084419

### User story
> _Describe the audience/user, enhancement or fix, and value / outcome desired._

**AS A Veteran or other user of VA.gov
**I WANT to be limited as to the length of my search text
**SO THAT I don't encounter an error when I kick off the search

### Engineering notes / background
* Explore all Search fields to determine what the current breakpoint is

## Acceptance criteria

**AC1: Unauthenticated user is searching on VA.gov**
**GIVEN**: I'm utilizing one of the Search fields on one of VA.gov's UNAUTHENTICATED pages
**WHEN:** I enter attempt to enter a string of text in the Search field that exceeds the point where an error will occur
**THEN:** the text I enter is truncated and I can't enter more text

**AC1: Authenticated user is searching on VA.gov**
**GIVEN:**  I'm utilizing one of the Search fields on one of VA.gov's on an AUTHENTICATED pages
**WHEN:** I enter attempt to enter a string of text in the Search field that exceeds the point where an error will occur
**THEN:** the text I enter is truncated and I can't enter more text

>Consider
> - Collab cycle requirements
> - Device sizes (mobile first)
> - Documentation updates / Change management
< - [Content model documentation](https://prod.cms.va.gov/admin/structure/cm_document)
> - Testing notes

- [ ]

**Comment by FranECross** [comment]
@thejordanwood Here are great questions from Randi regarding the behavior/AC that should be on this ticket:

- Truncate the text without error/warning
- Add a character counter by default and warn and truncate when it is exceeded
- door #3?
cc @randimays (thanks for tagging Jordan and the great questions!)

**Comment by randimays** [comment]
@FranECross Github tried to help with my `door #3` bullet - we should remove the link it auto-created to avoid confusion.

**Comment by thejordanwood** [comment]
@FranECross @randimays These are good questions Randi! I probably need to bring these to the design system team to get an answer. When would you need an answer by–next sprint?

Is there a way to allow for a larger character count or is our only option to truncate text/give an error message?

**Comment by FranECross** [comment]
@thejordanwood Re your question "When would you need an answer by–next sprint?" 

Ideally, yes, we'd like to groom and have ready for next sprint. Thanks! cc @randimays @jilladams

**Comment by randimays** [comment]
@thejordanwood I don't think a larger character count is our answer here.

Taking a look at the logs, it doesn't seem like this happens often. In the past month, this happened 62 times, and some (if not most) were bots. Others were folks who seem to misunderstand how to use the search engine and type very long questions or letters thinking they are reaching a person.

**Comment by aklausmeier** [comment]
@thejordanwood current error handling screenshot below, was able to reproduce

Since error alert already exists, possible just add a max character count using [error handling for text input in VA Design System](https://design.va.gov/components/form/text-input#error). 

![Screenshot 2023-11-01 at 1 49 29 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/137226108/4a7c5c8a-7958-4894-88bc-95debb130c69)

**Comment by thejordanwood** [comment]
@randimays @FranECross I can mockup and write the error message for this whenever we get to this ticket.

**Comment by jilladams** [comment]
Noting from convo with @FranECross : This ticket will be to implement the fix, and she will cut a UX ticket for the mockup / error message.

**Comment by randimays** [comment]
Noting here that I am awaiting an email reply from Search.gov about whether their API enforces a character count. We are receiving a 400 (I believe) from their API when we send strings that are too long. Once we know the character count (number), we can consistently enforce this across the front end and vets-api.

**Comment by randimays** [comment]
Should this issue have a blocked tag and be in the Blocked column?

**Comment by FranECross** [comment]
@randimays Thanks for the reminder! 😁 Just added the label and moved to Blocked.

**Comment by randimays** [comment]
Confirming here: Max character count allowable by Search.gov is 255.

**Comment by FranECross** [comment]
@randimays Thanks for the character count info! I updated the AC with that info, and also added 'Happy Path' AC just for completeness. cc @jilladams

**Comment by randimays** [comment]
Adding a note here that a VA.gov search also exists as an option on the [Resources & Support page](https://www.va.gov/resources/) (you can select the `All VA.gov` radio button).

**Comment by thejordanwood** [comment]
I created a mockup in [this Sketch file](https://sketch.com/s/433403fb-b16c-4c98-867e-181d86208ba2) that shows the error message we want to use.

**Comment by FranECross** [comment]
@randimays @jilladams Just want to confirm that I can move this ticket out of Blocked now (character count has been confirmed and Jordan has created a mockup in Sketch.) Thanks!

**Comment by chriskim2311** [comment]
@thejordanwood Question about the search button. I have allowed for functionality that will prevent a user submission until the search string is less than 255 characters error message will display when this happens. Do we want to show a disabled button color when this happens or just prevent the button from searching?

**Comment by thejordanwood** [comment]
@chriskim2311 I think it would be better to disable the button, since the search can't happen until it's less than 255 characters anyway. 

I'm going to tag @aklausmeier on this in case she has a different opinion.

**Comment by chriskim2311** [comment]
@thejordanwood Thanks! Should we change the color of the button as well to show that its disabled. Right now we just make the button unclickable. Here is the PR env to take a look thanks!

http://f4cde8b5356f8169942fe512c48359ab.review.vetsgov-internal/

**Comment by thejordanwood** [comment]
@chriskim2311 Yes, the color of the button should use the [disabled button styling](https://design.va.gov/storybook/?path=/docs/components-va-button--primary#disabled) that can be seen in the design system storybook.

**Comment by chriskim2311** [comment]
This ticket will carry over to next sprint. Have some pending a11y changes that need to be completed

**Comment by FranECross** [comment]
* Don't restrict field length
* Don't disable the button
* Add screen reader text
* Don't send error to the error reporting system

@chriskim2311

**Comment by FranECross** [comment]
@laflannery @chriskim2311 @jilladams I've updated the AC and my comment above from the meeting. Please let me know if you have any questions or if anything is confusing. Thanks!

**Comment by jilladams** [comment]
Tested in prod: submit a search with 256 chars, get an error screen: 
![Screenshot 2023-12-18 at 4 04 21 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/85581471/52c4f405-fdd5-44f5-9f2f-dc46c4764154)

Noting: the whole form gets red error highlighting, due to the .error class on the div that contains the alert + search form. I believe that's a design system component issue, but we can open a new ticket to address if that's something we need to address / revise. @chriskim2311 mind taking a look at that markup / css and comment here whether that's adjustable/needs a new ticket, or feedback that should go to the DS team for consideration with the component? (If this is, indeed, a DS component?)

Submit with 255 or less, get results.

**Comment by jilladams** [comment]
![Screenshot 2023-12-18 at 4 06 34 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/85581471/89a362b9-4adf-46b5-b811-36e78a0fd993)

**Comment by jilladams** [comment]
CSS fix is in PR, and is slightly out of scope for this ticket anyway. So: closing.
